guava/src/com/google/common/collect/ImmutableSet.java

```diff
@@ -91,7 +91,7 @@
    * the first are ignored.
    */
   public static <E> ImmutableSet<E> of(E e1, E e2) {
-    return construct(2, e1, e2);
+    return construct(2, Iterators.forArray(e1, e2));
```

Comment:
I think it makes sense to use the known number of elements as a sizing hint for the of(...) methods.